tests/unittests/test_net.py

```diff
@@ -4899,6 +4899,18 @@ def test_gi_excludes_bridges(self):
              mock.call('b1')],
             any_order=True)
 
+    def test_gi_excludes_bkacklist_drivers(self):
+        self._mock_setup()
+        blacklist = ['mlx4_core', 'mlx5_core']
```

Comment:
I'm not sure this is adding anything useful as we already know that calling the net.get_interfaces filters by blacklist.  Rather we need a test that confirms that the Azure Datasource updates the distro.networking object with the correct blacklist from the datasource.  And further that AzureDatasource.network_config output does not attempt to configure devices with drivers that are on the blacklist.

Reply:
Thanks. I have added a test to verify that the datasource Azure correctly updates the distro.networking object with its blacklist and also that the eventual call to net.get_interfaces does have the right blacklist passed in to it. The generated network_config from Azure already has a test to make sure it calls to generate_fallback_config with the blacklist (test_fallback_network_config). Happy to update that particular test if you think it's not sufficient.